packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/webVitalsUtils.ts

```diff
@@ -0,0 +1,252 @@
+import type { SlProgressRing } from "astrolace:types";
+import type { WebVitalsResponseItem } from "studiocms-dashboard:web-vitals";
+
+// Misc Utils
+export function msToSeconds(ms: number): number {
+    return ms / 1000;
+}
+
+// CLS Calculations
+export function calculateClsAverage(clsValues: number[]): number {
+
+    const sum = clsValues.reduce((acc, curr) => acc + curr, 0);
+    const average = sum / clsValues.length;
+    
+    // Round average to two decimal places
+    return Math.round(average * 100) / 100;
+}
+
+export const clsDataAverage = (webVitalData: WebVitalsResponseItem[]) => {
+    const clsData: number[] = [];
+    if (webVitalData) {
+        for (const item of webVitalData) {
+            if (item.name === "CLS") {
+                clsData.push(item.value);
+            }
+        }
+    }
+    return calculateClsAverage(clsData);
+};
+
+export function calculateClsScoreText(cls: number): string {
+    if (cls <= 0.1) {
+        return "Excellent"; // Excellent
+    }if (cls <= 0.25) {
+        return "Good"; // Good to Excellent
+    }if (cls <= 0.5) {
+        return "Fair"; // Fair to Good
+    }
+    return "Poor"; // Poor to Fair
+}
+
+export function calculateClsScorePercent(cls: number): number {
+    if (cls <= 0.1) {
+        return 100; // Excellent
+    }if (cls <= 0.25) {
+        return Math.round(100 - ((cls - 0.1) / (0.25 - 0.1)) * 25); // Good to Excellent
```

Comment:
Do we have a link for where we're getting these calculations from? Or a comment to explain what's happening here? Just a lot of magic numbers

Reply:
https://web.dev/articles/cls#what-is-a-good-cls-score

Reply:
[2e5f04a](https://github.com/astrolicious/studiocms/pull/125/commits/2e5f04a92da69eb44ba45385409c4d99014b2878) comments